### PR TITLE
MNT-17706: Bulk Import: Error message does not give hints about the f…

### PR DESCRIPTION
**Pull request by mihamimi**
Bulk Import: Error message does not give hints about the file triggering the error

Catch the exception and throw new one that appends the metadata file path and the cause. In case of an exception, the file path will be displayed in alfresco.log and in bulk import status page.

### CODE REVIEW

File: src/main/java/org/alfresco/repo/bulkimport/metadataloaders/AbstractMapBasedMetadataLoader.java [hunk 1]
```diff
@@ -101,14 +101,27 @@ public final String getMetadataFileExtension()
...
+            }
+            catch (Exception e)
+            {
+                log.error("Error encountered when reading metadata file '" + metadataFilePath + "'.");
+                throw new RuntimeException("Exception from reading file: '" + metadataFilePath + "'.", e);
+            }
+        }
+    }
 
+    private void loadMetadataInternal(Metadata metadata, Path metadataFile, String metadataFilePath)
```
Comment (andrei-rebegea) [inline src/main/java/org/alfresco/repo/bulkimport/metadataloaders/AbstractMapBasedMetadataLoader.java]: the ideal number of parameters to a method is 0. You can easily get this "metadataFilePath" information inside the method, from the other variables.

continue to add final to the remaining parameters

File: src/main/java/org/alfresco/repo/bulkimport/metadataloaders/AbstractMapBasedMetadataLoader.java [hunk 2]
```diff
@@ -101,14 +101,27 @@ public final String getMetadataFileExtension()
...
             final Path metadataFile = contentAndMetadata.getMetadataFile();
+            final String metadataFilePath = FileUtils.getFileName(metadataFile);
+            try
+            {
+                loadMetadataInternal(metadata, metadataFile, metadataFilePath);
+            }
+            catch (Exception e)
+            {
+                log.error("Error encountered when reading metadata file '" + metadataFilePath + "'.");
+                throw new RuntimeException("Exception from reading file: '" + metadataFilePath + "'.", e);
```
Comment (andrei-rebegea) [inline src/main/java/org/alfresco/repo/bulkimport/metadataloaders/AbstractMapBasedMetadataLoader.java]: use org.alfresco.error.AlfrescoRuntimeException

File: src/main/java/org/alfresco/repo/bulkimport/metadataloaders/AbstractMapBasedMetadataLoader.java [hunk 3]
```diff
@@ -101,14 +101,27 @@ public final String getMetadataFileExtension()
...
+            }
+        }
+    }
 
+    private void loadMetadataInternal(Metadata metadata, Path metadataFile, String metadataFilePath)
+    {
```
Comment (andrei-rebegea) [inline src/main/java/org/alfresco/repo/bulkimport/metadataloaders/AbstractMapBasedMetadataLoader.java]: i think this block is no longer indented properly. make a separate commit to fix the indent.

**Comment by andrei-rebegea** [comment]
Looks great!

File: src/main/java/org/alfresco/repo/bulkimport/metadataloaders/AbstractMapBasedMetadataLoader.java [hunk 4]
```diff
@@ -101,68 +101,85 @@ public final String getMetadataFileExtension()
...
                 }
             }
-            else
-            {
-                if (log.isWarnEnabled()) log.warn("Metadata file '" + FileUtils.getFileName(metadataFile) + "' is not readable.");
-            }
+        }
+        else
+        {
+            if (log.isWarnEnabled())
```
Comment (derekhulley) [inline src/main/java/org/alfresco/repo/bulkimport/metadataloaders/AbstractMapBasedMetadataLoader.java]: We still need to use braces, please.
Please fix on {{develop}}.

Comment (mihamimi) [reply src/main/java/org/alfresco/repo/bulkimport/metadataloaders/AbstractMapBasedMetadataLoader.java]: https://github.com/Alfresco/alfresco-repository/commit/1d56cf1b7cf715e70cbe1f580af164697f572901 done
